Add downloaded wal-g file check and some other adjustments

Jamal-B opened the pull request:
## Description

- Add wal-g download checksum check to avoid installing a corrupted or altered version. Moreover, the unarchiving task is adapted to the fact that the files included in the tar.gz of recent versions of wal-g have changed compared to older versions.
- Remove wal-g `always` ansible tags: we don't want ansible to always execute the wal-g role if we are targeting another role of the playbook.
- Set files's mode as advised by ansible documentation.

## Tests
- Change the wal-g version to upgrade or downgrade it while replacing the corresponding sha256 checksum, then validate that the desired version is correctly downloaded and installed.
- Repeat the same test but without updating the checksum and validate that an ansible error is thrown.
- Run ansible on a different tag than the one used to include the wal-g role and validate that the wal-g role is no longer executed.

Jamal-B committed:
Remove wal-g always ansible tags

we don't want ansible to always execute the wal-g role if we are
targeting another role of the playbook.

Jamal-B committed:
Verify wal-g download checksum

Moreover, the unarchiving task is adapted to the fact that the files
included in the tar.gz of recent versions of wal-g have changed
compared to older versions.

Jamal-B commented on tasks/main.yml:
```diff
@@ -92,6 +92,17 @@
+  changed_when: false
+  loop:
+    - root
+    - postgres
```
Même si au quotidien on ne va pas toutes les utiliser, la liste des commandes wal-g ...
```
usage:
wal-g [command]

Available Commands:
backup-fetch  Fetches a backup from storage
backup-list   Prints available backups
backup-mark   Marks a backup permanent or impermanent
backup-push   Makes backup and uploads it to storage
catchup-fetch Fetches an incremental backup from storage
catchup-list  Prints available incremental backups
catchup-push  Creates incremental backup from lsn
completion    Output shell completion code for the specified shell
copy          copy specific or all backups
delete        Clears old backups and WALs
flags         Display the list of available global flags for all wal-g commands
help          Help about any command
pgbackrest    Interact with pgbackrest backups (beta)
st            (DANGEROUS) Storage tools
wal-fetch     Fetches a WAL file from storage
wal-push      Uploads a WAL file to storage
wal-receive   Receive WAL stream with postgres Streaming Replication Protocol and push to storage
wal-restore   Restores WAL segments from storage.
wal-show      Show storage WAL segments info grouped by timelines.
wal-verify    Verify WAL storage folder. Available checks: integrity, timeline.
```

est bien plus fournie que la liste des wrappers créés par le rôle :
```
ls -1 /usr/local/scripts/

walg-backup.sh
walg-fetch.sh
walg-list.sh
walg-push.sh
walg-restore.sh
walg-rotate.sh
```

Et je trouvais que pour faire par exemple du debug ou intervenir en astreinte, c'était bien plus pratique (que ce soit sous le user root ou postgres) de faire `wal-g <command>` plutôt que `wal-g --config /etc/default/wal-g.env <command>`.

Jamal-B replied:
Si le but est de ne pas diverger du rôle source, alors ok pourquoi pas.

Jamal-B commented on tasks/main.yml:
```diff
@@ -92,6 +92,17 @@
+  changed_when: false
+  loop:
+    - root
+    - postgres
```
Oui, ce sont ces noms de fichiers (`$HOME/.walg.env`) que cherche `wal-g` si on ne lui passe pas de paramètre `--config`.

desbouis replied:
ok, en fait je pensais qu'il y avais une erreur dans les chemins, mais c'est moi l'erreur : je n'avais jamais utilisé la syntaxe `~<username>` :smile:

Jamal-B commented on tasks/main.yml:
```diff
@@ -121,4 +121,4 @@
     state: "{{ walg_cron_enabled | ternary('present', 'absent') }}"
   tags:
-    - always
+    - postgresql
```
Le rôle sera en effet appelé via les deux tags (je sais pas encore si c'est utile d'avoir les deux) `wal-g`  et `postgresql` depuis les meta du rôle postgresql.
J'ai supprimé `always` dans [ce commit](https://github.com/mediapart/ansible-wal-g/pull/1/commits/7a6f08a8dda27225778a87da94b396ac16c68779) directement tassé pour pouvoir le renommer.

Jamal-B commented on templates/walg-backup.sh.j2:
```diff
@@ -12,4 +12,4 @@ set -o allexport
 set +o allexport
 
-{{ walg_binary }} backup-push {{ walg_data_path }}
+{{ walg_binary }} backup-push {{ walg_data_path }} {{ walg_backup_options }}
```
J'ai fait [un fixup](https://github.com/mediapart/ansible-wal-g/pull/1/commits/40c3cb78c3c807c532abefe68ba665fab75a9f06) pour que ce soit plus clair.

desbouis replied:
ok merci.
En fait, je pense qu'on pourrait passer les options via les vars d'env correspondantes, par ex. pour `--verify`, il existe `WALG_VERIFY_PAGE_CHECKSUMS`. Et ces vars d'env sont à charger avec le `source {{ walg_env }}`.

Jamal-B replied:
T'as proposition est bonne ; mais tu as pris la plus simple des deux options :slightly_smiling_face: ... l'autre, `--full`, n'a pas d'équivalent en var d'env (pas trouvée dans la doc ni directement dans le code de wal-g, mais peut-être que je cherche mal).

Jamal-B committed:
Set files mode as advised by ansible documentation

https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html#parameter-mode

> For consistent results, quote octal numbers (for example, '644'
or '1777') so Ansible receives a string and can do its own conversion
from string into number. Adding a leading zero (for example, 0755)
works sometimes, but can fail in loops and some other circumstances.
